backend/app/services/adapters/workspace_archive.py

```
@@ -0,0 +1,429 @@
+            # Update task record
+            task.workspace_archived_at = datetime.now()
+            task.workspace_archive_key = archive_key
+            db.commit()
+
+            logger.info(
+                f"Successfully archived workspace for task {task_id} "
+                f"(size: {archive_size_mb:.2f}MB, key: {archive_key})"
+            )
+            return True, None
+
+        except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)